csrc/mmdeploy/operation/cuda/crop.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) OpenMMLab. All rights reserved.
+
+#include <cuda_runtime.h>
+
+#include "mmdeploy/core/utils/formatter.h"
+#include "mmdeploy/operation/vision.h"
+
+namespace mmdeploy::operation::cuda {
+
+namespace impl {
+
+template <typename T, int channels>
+void Crop(const T* src, int src_w, T* dst, int dst_h, int dst_w, int offset_h, int offset_w,
+          cudaStream_t stream);
+
+}
+
+class CropImpl : public Crop {
+ public:
+  Result<void> apply(const Tensor& src, Tensor& dst, int top, int left, int bottom,
+                     int right) override {
+    auto cuda_stream = GetNative<cudaStream_t>(stream());
+    auto desc = src.desc();
+
+    int h = bottom - top + 1;
+    int w = right - left + 1;
+    int c = desc.shape[3];
+    auto type = desc.data_type;
+
+    TensorShape shape{1, bottom - top + 1, right - left + 1, src.desc().shape[3]};
+    TensorDesc dst_desc{device(), src.desc().data_type, shape, desc.name};
+    Tensor dst_tensor{dst_desc};
+
+    if (DataType::kINT8 == type) {
```

Comment:
How about dispatch according to `type`?

Reply:
It's just 4 branches. I welcome anyone who is not me to refactor it.